src/main/java/com/tmh/controller/CartController.java

```diff
@@ -0,0 +1,167 @@
+	public String showProductDetail(@PathVariable("id") int id, Model model) {
+		logger.info("show product detail");
+	  
+	 	model.addAttribute("product", productService.findById(id));
+	  
+	 	return "views/client/productDetail";
+	}
+	
+	@RequestMapping(value = "/cart")
+	public String showCart(Model model, @ModelAttribute("cart") Cart cart) {
+		logger.info("show cart");
+
+		if (cart != null) {
+			model.addAttribute("cart", cart);
+		} else {
+			model.addAttribute("cart", new Cart());
+		}
+
+		return "views/client/cart";
+	}
+
+	@RequestMapping(value = "/add-to-cart", method = RequestMethod.POST)
+	public String addToCart(@ModelAttribute("cart") Cart cart, @RequestParam("productId") int productId,
+			@RequestParam("quantity") int quantity, RedirectAttributes redirectAttributes) {
+		logger.info("add to cart");
+
+		if (cart == null) {
+			cart = new Cart();
+		}
+
+		cart.addProduct(productService.findById(productId), quantity);
+		redirectAttributes.addFlashAttribute("cart", cart);
+
+		return "redirect:/cart";
+	}
+
+	@RequestMapping(value = "/update-cart", method = RequestMethod.POST)
+	public String updateCart(@ModelAttribute("cart") Cart cart, RedirectAttributes redirectAttributes) {
+		logger.info("update cart");
+
+		redirectAttributes.addFlashAttribute("cart", cart);
+
+		return "redirect:/cart";
+	}
+	
+	@RequestMapping(value = "/cart/{id}/remove")
+	public String deleteCart(@ModelAttribute("cart") Cart cart, @PathVariable("id") int id, RedirectAttributes redirectAttributes) {
+		logger.info("remove products in cart");
+		
+		cart.removeProduct(id);
+
+		redirectAttributes.addFlashAttribute("cart", cart);
+
+		return "redirect:/cart";
+	}
+	
+	@RequestMapping(value = "/create-order")
+	public String submitOrder(@ModelAttribute("cart") Cart cart, Model model) {
+		logger.info("create order");
+		
+		Order order = new Order();
+		
+		model.addAttribute("cart", cart);
+		model.addAttribute("user", getCurrentUser());
+		model.addAttribute("order", order);
+		
+		return "views/client/order";
+	}
+	
+	@RequestMapping(value = "/submit-checkout", method = RequestMethod.POST)
+	public String submitCheckout(@ModelAttribute("order") Order order, BindingResult bindingResult, @ModelAttribute("cart") Cart cart, RedirectAttributes redirectAttributes) {
+		logger.info("submit checkout");
+		
+		order.setUser(getCurrentUser());
+		order.setTotal(cart.getAmountTotal());
+		order.setStatus(0);
+		order.setIsDeleted(0);
+		order.setCreateOrderDate(LocalDateTime.now());
+		
+		orderService.saveOrUpdate(order);
+		
+		Order mOrder = orderService.findNotDeletedOrders().get(orderService.findNotDeletedOrders().size() - 1);
+		
+		for (CartItem cartItem : cart.getCartItems()) {
+			OrderItem orderItem = new OrderItem();
+			orderItem.setProduct(cartItem.getProduct());
+			orderItem.setOrder(mOrder);
+			orderItem.setQuantity(cartItem.getQuantity());
+			orderItem.setIsDeleted(0);
+			
+			orderItemService.saveOrUpdate(orderItem);
```

Comment:
code for này em bỏ vào service, sau khi gọi save order đồng thời save order item

chứ em để ri, trường hợp orderItem save ko thành công, thì order có rollback dc mô, vì 2 xử lý dc thao tác bởi 2 transaction khác nhau

Reply:
Dạ giống như cái Service hôm trước làm ở phần Admin ấy hả chị? Để em sửa lại ạ